### ISSUE TITLE
Sharky 775 unresponsive

### ISSUE THREAD

**Issue by xawill** [mh-g/ultramess#1]
Hi @mh-g,

Thank you very much for sharing this work!

I also have a Sharky 775 (Amtron Sonic D) I am trying to access through the optical ZVEI interface, but it looks like I cannot manage to wake the meter up. It is not sending anything.

Do you have any advice? Thanks a lot in advance.

BR,
Xavier

**Comment by mh-g** [comment]
Hi Xavier,

take a deeper look at lines 45-58 of the current revision of main.py. They are the result of quite some frustration with the very same problem that you seem to have. 

The technical reference states that for 2.2s a 0/1 toggled bit pattern has to be sent at 2400 baud, 8N1. This would mean to send **480 bytes** of hex 0x55. Indeed I had to send **528 bytes** of hex 0x55. Afterwards you are supposed to wait for 11 to 330(!!!) bits duration before starting with the regular communication (I chose 170 bits as the middle ground). You may want to analyse the transmitted sequence using a scope. Sometimes the real waveform does not really resemble the commanded one ...

Also note that between the trigger sequence and the command sequence, I need to change serial line setup. You may also want to analyze the received signal using a scope ...

My self-made sensor head also had some difficulties with Tx/Rx crosstalk. In the end, I redesigned the casing I found on thingiverse.com. I printed everything using black filament and made sure to have an opaque wall between the Tx emitter and Rx diode. (Having two sensor heads to test communications by putting them against each other 180 degrees rotated may help to diagnose such problems.)

Good luck and BR,
mh-g

**Comment by xawill** [comment]
Thanks a lot for taking the time to share your very valuable experience! I'll follow your advice, try again and eventually get hold of an oscilloscope :)

**Comment by xawill** [comment]
Hi @mh-g,

FYI, I successfully managed to read data from the Sharky 775 after trying again following your advice (without oscilloscope). Thanks a lot!

I am not quite sure what exactly did the trick, but I had to use 480 0x55 bytes (instead of the 528 you have). And I also had to fix what I think is a typo in the code line 55: the waiting time should be 2.2s + "11-330 bit time" (instead of the 1.2s you have).

It seems indeed that this is all a question of timing. And since I am not using the same hardware as yours, probably this explains the discrepancy.

Anyway, thank you for motivating me to try again; successfully this time.

Cheers,
Xavier
